Featurebranch - HTML Form

OlukaDenis opened the pull request:
We're done with implementing project two features and we would request a code review.
Thanks!

OlukaDenis committed:
Configured the stickler, added the images and project initial files

Grifo89 committed:
Finished HTML and start css

Grifo89 committed:
little changes

OlukaDenis committed:
Fixed the inner form alignment

Grifo89 committed:
small changes

Grifo89 committed:
fixing git three

Grifo89 committed:
More Styling stuff

OlukaDenis committed:
Added more styling

Grifo89 committed:
Finished styling

OlukaDenis committed:
More styling and added more content to the Readme file

bolah2009 commented on style.css:
```diff
@@ -0,0 +1,210 @@
+
+* {
+  box-sizing: border-box;
+  /* font-family: AvenirNext, Avenir, "Helvetica Neue", Helvetica, Arial,
```
#### `REQUIRED`

Consider deleting commented out codes. If you need to review a previous version of your code, use git history.

bolah2009 commented on index.html:
```diff
@@ -0,0 +1,110 @@
+                <label for="">
+                  <p>Email address</p>
+                  <input type="email" name="email" />
+                </label>
```
#### `HINT`

:bulb: Consider associating the `<label>` with the `<input>` element, to do this you need to give the `<input>` an `id` attribute. The `<label>` then needs a `for` attribute whose value is the same as the input's id. 
- Example 1: 

``` html
  <label for="email">Email address</label> <br>
  <input type="email" name="email" id="email">
```
- Example 2: Alternatively, you can nest the `<input>` directly inside the `<label>` like this: 

``` html
  <label for="email">Email address <br>
       <input type="email" name="email" id="email">
  </label> 
```

> #### Associating a `<label>` with an `<input>` element offers some major advantages:
> 
> - The label text is not only visually associated with its corresponding text input; it is programmatically associated with it too. This means that, for example, a screen reader will read out the label when the user is focused on the form input, making it easier for an assistive technology user to understand what data should be entered.
> 
> - You can click the associated label to focus/activate the input, as well as the input itself. This increased hit area provides an advantage to anyone trying to activate the input, including those using a touch-screen device.

You can learn more [here](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/label)

bolah2009 commented on index.html:
```diff
@@ -0,0 +1,110 @@
+            </div>
+            <form class="form" action="#" method="post">
+              <fieldset class="inputs">
+                <label for="">
```
Let's give `<input type="email" name="email" />` an `id` attribute, then use it as a value for the `for` attribute here.
```suggestion
                <label for="<id of email input>">
```
See detailed explanation above.

bolah2009 commented on README.md:
```diff
@@ -1,2 +1,9 @@
+
+- [Denis Oluka](https://github.com/OlukaDenis)
+
+- [Christian Salazar](https://github.com/Grifo89)
```
Let's make the `README` file more descriptive, details like what was done in the projects, what was learned, link to the assignment, clone and source page, screenshots of the clone page can make README look more professional.
 > Remember about descriptive README file.

bolah2009 reviewed (changes requested):
# Nice job team :smile: 

### The [clone](https://raw.githack.com/OlukaDenis/HTML_Form/e3781906963ed11f9251a060cd8be37c04b762ff/index.html) looks good to me, uses HTML input attributes effectively. :clap: : 

However, there are a few changes required to be made before a :+1: 
- Fix linter and [HTML validation](https://validator.w3.org/nu/?doc=https%3A%2F%2Fraw.githack.com%2FOlukaDenis%2FHTML_Form%2Fe3781906963ed11f9251a060cd8be37c04b762ff%2Findex.html) errors/warnings.
  > Validate your HTML markup with https://validator.w3.org/#validate_by_upload.
- Delete commented out codes
  > Inline comments: It’s better to delete commented code. If you need to review a previous version of your code, use git history.
- Make `README` file more descriptive
  > Remember about descriptive README file.
- Let's make this Pull Request look more professional. [See below](https://github.com/OlukaDenis/HTML_Form/pull/1#pullrequestreview-291475082)
  - [NEXT PROJECT]: Use meaningful commits messages. [Learn more here](https://www.freecodecamp.org/news/level-you-up-to-awesome-commit-messages-a85558cb90e8/)
  - [REQUIRED] Use meaningful names for  Pull Requests.
  - [REQUIRED] Add a short summary of changes in Pull Request description.
- [NEXT PROJECTS]: For next (HTML/CSS) project, please follow `GitHub flow` [guidelines](https://microverse.pathwright.com/library/microverse-onboarding/86777/path/step/53987922/). What is missing here is a descriptive branch name instead of `featurebranch`, think of feature branch as a branch name that is descriptive enough to have an idea of what the branch is used for.


#### :recycle:  Make these changes and submit another code review request. :+1: 

#### Feel free to reach me on slack if you having issues implementing these features. :+1: 

Good luck with your next submission. :star2: 

Cheers
[Bola](https://bolabuari.com/)

###### PS: Note that comments tagged as `OPTIONAL` are not part of the minimum requirements to approve this PR

bolah2009 reviewed (commented):
# Rename PR

  ![rename PR](https://user-images.githubusercontent.com/36057474/63838245-ec569f80-c974-11e9-93a9-83a4db5c259f.gif)

-----

# Update PR description

  ![update-PR-description-html](https://user-images.githubusercontent.com/36057474/64305800-6e078800-cf88-11e9-82f0-04e13fd6cb8d.gif)

OlukaDenis committed:
Updates for Project Review:
1. Removed the commented code from the CSS file
2. Resolved the label ID issue
3. Removed the <p> from the label
4. Added more details in the README file
5. Added the app image
6. Passed the HTML validator with no errors

OlukaDenis committed:
Update README.md

samgaco commented on style.css:
```diff
@@ -0,0 +1,212 @@
+nav p {
+  font-size: 12px;
+  color: #505050;
+  padding: 3px 0px;
```
Correct the linter checks, for example here `Unexpected unit (length-zero-no-unit) [error]` is telling you that 0 doesn't need units (px, rem, etc)

samgaco commented on style.css:
```diff
@@ -0,0 +1,212 @@
+  height: 44px;
+  border: none;
+
+  color: white;
```
check the linter fails, for example here is failing because of the white space. You could use some CSS [formatter](https://codebeautify.org/css-beautify-minify/21ce48) to be faster with that.

samgaco reviewed (changes requested):
Hi @Grifo89 and @OlukaDenis :high_brightness: 

You have to solve **all the[ linter failing checks](https://github.com/OlukaDenis/HTML_Form/pull/1/checks?check_run_id=231286804)** before being able to proceed. Submit back for another code review when you are done with that :+1: 

------------------------
* ###### STATUS OF THE PROJECT: REQUESTED CHANGES :stop_sign:

Grifo89 committed:
This commit is for css reset change. I deleted a file which we used to
reset browser and changed for an other one shorter and put it into the
main CSS file.

OlukaDenis committed:
Beatified the CSS:
1. Removed the imports from the CSS
2. Beautified the CSS online

OlukaDenis committed:
Resolved the stickler errors front the CSS file

OlukaDenis committed:
Resolved the stickler errors front the CSS file

OlukaDenis committed:
Resolved the stickler errors front the CSS file

OlukaDenis commented:
Hello there,
We have resolved all the stickler errors and our code is ready for the final review. 
Thanks

OlukaDenis committed:
Update README.md

ebukaume commented:
Hi @OlukaDenis and @Grifo89 

You guys have really come a long way in this project. :v: 

Kindly, remove the comments in the style.css file. Yes, they are not codes but generally; you don't need those comments.

**After that, you can go ahead and merge! Congratulations :clap:** 

Good luck with your next projects!

OlukaDenis committed:
Project Complete:
1. Removed the comments from the style.css file
2. Passed all the code reviews

OlukaDenis committed:
Merge branch 'featurebranch' of github.com:OlukaDenis/HTML_Form into featurebranch

OlukaDenis commented:
1. Passed all the stickler checks
2. Passed the code reviews
3. Project declared done
